google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java

```diff
@@ -137,6 +139,46 @@
   private static final Entity ENTITY3 = Entity.newBuilder(ENTITY1).setKey(KEY3).remove("str")
       .set("null", NULL_VALUE).set("partial1", PARTIAL_ENTITY2).set("partial2", ENTITY2).build();
 
+  private <T> Iterator<T> getStronglyConsistentResults (Query scQuery, Query query) {
+    QueryResults<T> scResults = DATASTORE.run(scQuery);
+    List<T> scResultsCopy = makeResultsCopy(scResults);
+    Set<T> scResultsSet = new HashSet<>(scResultsCopy);
+
+    infiniteloop:
+    while(true) {
+      QueryResults<T> results = DATASTORE.run(query);
+      List<T> resultsCopy = makeResultsCopy(results);
+      if (!haveSameSize(scResultsCopy.iterator(), resultsCopy.iterator())) {
```

Comment:
This method takes linear time to compare two collections, when it is not necessary here as both scResultsCopy and resultsCopy are lists. Please replace it with just constant time and simpler to understand version:
```
if(csResultsCopy.size() != resultsCopy.size()) {
    continue;
}
```